src/lkl/setup.c

```diff
@@ -697,13 +681,15 @@ static void lkl_mount_disk(
         dev_str = dev_str_enc;
     }
 
-    if ((lkl_trace_lkl_syscall_bak && !sgxlkl_trace_lkl_syscall) ||
-        (lkl_trace_internal_syscall_bak && !sgxlkl_trace_internal_syscall))
+    if ((lkl_trace_lkl_syscall_bak && !tcfg->lkl_syscall) ||
+        (lkl_trace_internal_syscall_bak && !tcfg->internal_syscall))
```

Comment:
This check is also redundant.  You can unconditionally reset the `sgxlkl_enclave_state.trace_enabled.lkl_syscall` to `sgxlkl_enclave_state.config->trace.lkl_syscall` and the same for the internal variant.

Reply:
(Same as above, avoided emitting the spurious warning.)